scheduler/pkg/klcpermit/workflow_manager.go

```diff
@@ -54,73 +56,96 @@ type Manager interface {
 type WorkloadManager struct {
 	dynamicClient dynamic.Interface
 	Tracer        trace.Tracer
+	bindCRDSpan   map[string]trace.Span
 }
 
 func NewWorkloadManager(d dynamic.Interface) *WorkloadManager {
 	sMgr := &WorkloadManager{
 		dynamicClient: d,
 		Tracer:        otel.Tracer("keptn/scheduler"),
+		bindCRDSpan:   make(map[string]trace.Span, 100),
 	}
 	return sMgr
 }
 
-var bindCRDSpan = make(map[string]trace.Span, 100)
+func (sMgr *WorkloadManager) ObserveWorkloadForPod(ctx context.Context, handler framework.WaitingPod, pod *corev1.Pod) {
+	factory := dynamicinformer.NewFilteredDynamicSharedInformerFactory(sMgr.dynamicClient, 0, pod.GetNamespace(), nil)
 
-func (sMgr *WorkloadManager) Permit(ctx context.Context, pod *corev1.Pod) Status {
-	//List workloadInstance run CRDs
-	name := getCRDName(pod)
-	crd, err := sMgr.GetCRD(ctx, pod.Namespace, name)
+	gvr, _ := schema.ParseResourceArg("keptnworkloadinstances.v1alpha1.lifecycle.keptn.sh")
```

Comment:
why don't you directly use  workloadInstanceResource from line 22 ?

Reply:
great catch!